gcore/gdal_misc.cpp

```diff
@@ -5536,6 +5640,30 @@ double GDALGetNoDataReplacementValue(GDALDataType dt, double dfNoDataValue)
         else
             dfReplacementVal = dfNoDataValue + 1;
     }
+    else if (dt == GDT_Float16)
+    {
+
+        if (GDALClampDoubleValue(dfNoDataValue,
+                                 std::numeric_limits<GFloat16>::lowest(),
+                                 std::numeric_limits<GFloat16>::max()))
+        {
+            return 0;
+        }
+
+        // TODO(Schnetter): Why should we modify the noDataValue?
```

Comment:
See https://github.com/OSGeo/gdal/pull/11713

Reply:
Thanks, I implemented this.